src/abstractdataframe/reshape.jl

```diff
@@ -170,7 +170,7 @@ function unstack(df::AbstractDataFrame, rowkey::Int, colkey::Int, value::Int)
     Nrow = length(refkeycol.pool)
     Ncol = length(keycol.pool)
     # TODO make fillNA(type, length)
-    payload = DataFrame(Any[DataArray(fill(valuecol[1], Nrow), fill(true, Nrow)) for i in 1:Ncol], map(@compat(Symbol), keycol.pool))
+    payload = DataFrame(Any[DataArray(eltype(valuecol), Nrow) for i in 1:Ncol], map(@compat(Symbol), keycol.pool))
```

Comment:
Passes without @compat(Symbol) on 0.5.0-rc3.  I don't have any easy way to check 0.4.6 offline but can push this and see how the build goes.